src/kernels/kernelStartupCodeProvider.node.ts

```diff
@@ -2,80 +2,43 @@
 export class KernelStartupCodeProvider implements IStartupCodeProvider {
     public priority = StartupCodePriority.Base;
 
-    constructor(
-        @inject(IConfigurationService) private readonly configService: IConfigurationService,
-        @inject(IFileSystem) private readonly fs: IFileSystem,
-        @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService
-    ) {}
+    constructor(@inject(KernelWorkingFolder) private readonly kernelWorkingFolder: KernelWorkingFolder) {}
 
     async getCode(kernel: IKernel): Promise<string[]> {
-        // If this is a live kernel, we shouldn't be changing anything by running startup code.
+        // We cannot change paths for remote live kernels.
+        if (kernel.kernelConnectionMetadata.kind === 'connectToLiveRemoteKernel') {
+            return [];
+        }
+        // Only local kernels are supported.
         if (
-            !isPythonKernelConnection(kernel.kernelConnectionMetadata) &&
-            kernel.kernelConnectionMetadata.kind !== 'connectToLiveRemoteKernel'
+            !isLocalConnection(kernel.kernelConnectionMetadata) &&
+            !isLocalHostConnection(kernel.kernelConnectionMetadata)
         ) {
             return [];
         }
-        if (!isLocalConnection(kernel.kernelConnectionMetadata)) {
+        if (!isPythonKernelConnection(kernel.kernelConnectionMetadata)) {
             return [];
         }
 
-        if (
-            isLocalConnection(kernel.kernelConnectionMetadata) ||
-            isLocalHostConnection(kernel.kernelConnectionMetadata)
-        ) {
-            let suggestedDir = await calculateWorkingDirectory(
-                this.configService,
-                this.workspaceService,
-                this.fs,
-                kernel.resourceUri
-            );
-            if (suggestedDir && (await this.fs.exists(suggestedDir))) {
-                traceInfo('UpdateWorkingDirectoryAndPath in Kernel');
-                // We should use the launch info directory. It trumps the possible dir
-                return this.getChangeDirectoryCode(kernel, suggestedDir);
-            } else if (kernel.resourceUri && (await this.fs.exists(kernel.resourceUri))) {
-                // Combine the working directory with this file if possible.
-                suggestedDir = Uri.file(
-                    expandWorkingDir(
-                        getFilePath(suggestedDir),
-                        kernel.resourceUri,
-                        this.workspaceService,
-                        this.configService.getSettings(kernel.resourceUri)
-                    )
-                );
-                if (suggestedDir && (await this.fs.exists(suggestedDir))) {
-                    traceInfo('UpdateWorkingDirectoryAndPath in Kernel');
-                    return this.getChangeDirectoryCode(kernel, suggestedDir);
-                }
-            }
+        // If this is a live kernel, we shouldn't be changing anything by running startup code.
```

Comment:
Comment might be in the wrong place now?

Reply:
Small suggestion here, these checks are doubled up both here and in getWorkingDirectory. Maybe could just be in one place? If get working directory just returns undefined for remote or non-python then these checks here don't seem needed as a function prereq.

Reply:
Agreed, Fixed